src/components/gating/GatingSubscriber.js

```diff
@@ -0,0 +1,257 @@
+import React, { useMemo } from 'react';
+import TextField from '@material-ui/core/TextField';
+import BaseScatterplotSubscriber, {
+  BASE_SCATTERPLOT_DATA_TYPES,
+} from '../base-scatterplot/BaseScatterplotSubscriber';
+import { capitalize } from '../../utils';
+import {
+  useLoaders,
+  useCoordination,
+} from '../../app/state/hooks';
+import {
+  useReady, useUrls,
+} from '../hooks';
+import {
+  useCellsData,
+  useExpressionMatrixData,
+} from '../data-hooks';
+import OptionSelect from '../shared-plot-options/OptionSelect';
+import { useStyles } from '../shared-plot-options/styles';
+import { COMPONENT_COORDINATION_TYPES } from '../../app/state/coordination';
+
+
+/**
+   * A subscriber component for the gating scatterplot.
+   * @param {object} props
+   * @param {number} props.uuid The unique identifier for this component.
+   * @param {string} props.theme The current theme name.
+   * @param {object} props.coordinationScopes The mapping from coordination types to coordination
+   * scopes.
+   * @param {boolean} props.disableTooltip Should the tooltip be disabled?
+   * @param {function} props.removeGridComponent The callback function to pass to TitleInfo,
+   * to call when the component has been removed from the grid.
+   * @param {number} props.averageFillDensity Override the average fill density calculation
+   * when using dynamic opacity mode.
+   */
+export default function GatingSubscriber(props) {
+  const {
+    coordinationScopes,
+    observationsLabelOverride: observationsLabel = 'cell',
+  } = props;
+
+  // Get "props" from the coordination space.
+  const [{ dataset }] = useCoordination(
+    COMPONENT_COORDINATION_TYPES.scatterplot,
```

Comment:
```suggestion
    COMPONENT_COORDINATION_TYPES[Component.GATING],
```

(we set a bad example in the other `Subscriber` components, but we should be using these component constant values here)